---
title: C26111
ms.date: 11/04/2016
ms.topic: reference
f1_keywords:
- C26111
helpviewer_keywords:
- C26111
ms.assetid: 85fc740a-3bbb-41b8-a848-95e243a08da9
ms.openlocfilehash: 193c94442230b1afa5a45c5314fd6ead4f3a62e1
ms.sourcegitcommit: 7bea0420d0e476287641edeb33a9d5689a98cb98
ms.translationtype: MT
ms.contentlocale: fr-FR
ms.lasthandoff: 02/17/2020
ms.locfileid: "77417352"
---
# <a name="c26111"></a>C26111

> AVERTISSEMENT C26111 : l’appelant ne parvient pas à libérer le verrou \<Lock > avant d’appeler la fonction \<Func >.

L' `_Requires_lock_not_held_` d’annotation impose une condition préalable que le nombre de verrous pour le verrou spécifié ne peut pas être supérieur à zéro lorsque la fonction est appelée. L’avertissement C26111 est émis lorsqu’une fonction ne parvient pas à libérer le verrou avant d’appeler une autre fonction.

## <a name="example"></a>Exemple

L’exemple suivant génère l’avertissement C26111, car la condition préalable `_Requires_lock_not_held_` n’est pas respectée par l’appel à `DoNotLock` dans la section verrouillée.

```cpp
typedef struct _DATA
{
    CRITICAL_SECTION cs;
    int d;
} DATA;

_Requires_lock_not_held_(p->cs)

void DoNotLock(DATA* p)
{
    EnterCriticalSection(&p->cs);
    p->d = 0;
    LeaveCriticalSection(&p->cs);
}

void LockedFunction(DATA* p)
{
    EnterCriticalSection(&p->cs);
    DoNotLock(p); // Warning C26111
    LeaveCriticalSection(&p->cs);
}
```
